src/core-services/orders/util/getOrderQuery.js

```diff
@@ -5,35 +5,38 @@ import ReactionError from "@reactioncommerce/reaction-error";
  * @name getOrderQuery
  * @method
  * @memberof Order/helpers
- * @summary Creates Order mongo selector based on user permissions
+ * @summary Queries for an order and returns it if user has correct permissions
  * @param {Object} context An object containing the per-request state
- * @param {Object} selector A mongo selector
+ * @param {Object} selector Order ID or Reference ID to query
  * @param {String} shopId Shop ID of the order
  * @param {String} token An anonymous order token, required if the order was placed without being logged in
- * @returns {Object} A mongo selector
+ * @returns {Object} An order object
  */
 export async function getOrderQuery(context, selector, shopId, token) {
-  const { accountId: contextAccountId } = context;
-  const newSelector = { ...selector, shopId };
-  const userHasPermission = await context.userHasPermission(
+  const { collections } = context;
+
+  const order = await collections.Orders.findOne(selector);
```

Comment:
You are going to need to keep the `const newSelector = { ...selector, shopId };` line. Otherwise I could pass a `shopId` where I have permissions and an order ID that belongs to a different shop, and I'd still be able to see it.